client/jetpack-connect/plans.jsx

```diff
@@ -108,42 +119,25 @@ class Plans extends Component {
 		this.props.recordTracksEvent( 'calypso_jpc_help_link_click' );
 	};
 
-	isFlowTypePaid() {
-		return (
-			this.props.flowType === 'pro' ||
-			this.props.flowType === 'premium' ||
-			this.props.flowType === 'personal'
-		);
+	isFlowTypePaid( flowType ) {
+		return flowType === 'pro' || flowType === 'premium' || flowType === 'personal';
 	}
 
-	redirectToWpAdmin() {
-		if ( this.props.redirectingToWpAdmin ) {
-			return;
-		}
-
-		const { redirectAfterAuth } = this.props;
+	redirectToWpAdmin( props ) {
+		const { redirectAfterAuth } = props;
 		if ( redirectAfterAuth ) {
-			this.props.goBackToWpAdmin( redirectAfterAuth );
-		} else if ( this.props.selectedSite ) {
-			this.props.goBackToWpAdmin( this.props.selectedSite.URL + JETPACK_ADMIN_PATH );
+			props.goBackToWpAdmin( redirectAfterAuth );
+		} else if ( props.selectedSite ) {
+			this.props.goBackToWpAdmin( props.selectedSite.URL + JETPACK_ADMIN_PATH );
 		}
 		this.props.completeFlow();
 	}
 
 	redirect( path ) {
 		page.redirect( path + this.props.selectedSiteSlug );
-		this.redirecting = true;
 		this.props.completeFlow();
 	}
 
-	hasPreSelectedPlan() {
```

Comment:
Removing function entirely to make things simpler for using `nextProps`. Fairly arbitrary decision though, we could keep it if it helps readability.